### ISSUE TITLE
Pd: when stopping patch you get `make: *** [runide] Terminated`

### ISSUE THREAD

**Issue by giuliomoro** [BelaPlatform/Bela#521]
I guess cleanup needs improvement.

**Comment by giuliomoro** [comment]
This may be the Scope: it also happens on C++ projects containing a scope. Also, it is pretty intermittent and doesn't seem to get caught by `gdb`.

**Comment by giuliomoro** [comment]
I have done some tests. This has to do with `make` and its interactions with ad without the IDE.

The IDE seems to be sending `SIGTERM` to the `make` command used to start the project (which is something like `make '--no-print-directory' '-C' '/root/Bela/' 'runide' 'PROJECT=exampleTempProject' 'CL=-p16 -C8 -B16 -H-6 -N1 -G1 -M0 -D0 -A0 --pga-gain-left=10 --pga-gain-right=10  -X0'`).

When this happens, `make` seems to exit with one of:

```
Makefile:552: recipe for target 'runide' failed
make: *** [runide] Terminated
```
(first line is on `stdout`, second one is on `stderr`)
or 
```
make: *** wait: No child processes.  Stop.
make: *** Waiting for unfinished jobs....
make: *** wait: No child processes.  Stop.
```
(all on `stderr`)

The latter form seems to be more common. There seems to be no link between the amount of time it takes for the process to exit (`SIGTERM` is correctly propagated to the Bela process) and the one or the other outcome(although the former seems to happen more often with Pd projects). The latter outcome is currently always hidden from the user:

```
dev/src/Views/ConsoleView.js:183:		if (!log.includes('make: *** wait')) // block unneccesary errors when killing process
```

This behaviour is exactly the same when running the same `make` command in the console (or in the IDE) and delivering the signal with:
```
kill -15 $(printf " -%d\n" `ps aux | grep ma[k]e | cut --bytes 10-15`)
```
(although in the latter case there is one extra line with `Terminated` in both cases, maybe this is added by the shell? And how does it know about it?).

A fix here could be:

ignore all lines
```
Makefile:[...]: recipe for target 'runide' failed
```
from `stdout` and all lines
```
make: *** [runide] Terminated
```
from `stderr`.

However, I am not sure if this is hiding some potentially important messages.

**Comment by giuliomoro** [comment]
this may be because ,despite `WSServer` calls `server->terminate()` , we are not actually waiting for `server_task` to actually return. It may even need a nudge to unblock it.

**Comment by giuliomoro** [comment]
there was a whole mess in the cleanup, with on_disconnect callbacks called after the Scope/Gui object was destroyed and/or the WSServer destructor never actually called because of circular references and the refcount never going to zero (thanks smart pointers).

It was fixed this week with a combination of 544bab2652fb0a0018d05928a5ceea66ad766bc6 and 38b545f0526f7ac9def2f65adf068bb0d5591f8 and everything in between and around them
